src/Microsoft.DotNet.Watcher.Core/Internal/ProcessExtensions.cs

```diff
@@ -0,0 +1,111 @@
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using Microsoft.Extensions.PlatformAbstractions;
+
+namespace Microsoft.DotNet.Watcher.Core.Internal
+{
+    public static class ProcessExtensions
+    {
+        private static readonly bool _isWindows = PlatformServices.Default.Runtime.OperatingSystemPlatform == Platform.Windows;
+
+        public static void KillTree(this Process process, TimeSpan timeout)
+        {
+            string stdout;
+            if (_isWindows)
+            {
+                RunProcessAndWaitForExit(
+                    "taskkill",
+                    $"/T /F /PID {process.Id}",
+                    timeout,
+                    out stdout);
+            }
+            else
+            {
+                var childs = GetAllChildIdsUnix(process, timeout);
+                foreach (var childId in childs)
+                {
+                    KillProcessUnix(childId, timeout);
+                }
+            }
+        }
+
+        private static IEnumerable<int> GetAllChildIdsUnix(Process process, TimeSpan timeout)
+        {
+            var childs = new HashSet<int>();
+            GetAllChildIdsUnix(process.Id, childs, timeout);
+            return childs;
+        }
+
+        private static void GetAllChildIdsUnix(int parentId, ISet<int> childs, TimeSpan timeout)
+        {
+            string stdout;
+            var exitCode = RunProcessAndWaitForExit(
+                "pgrep",
+                $"-P {parentId}",
+                timeout,
+                out stdout);
+
+            if (exitCode == 0 && !string.IsNullOrEmpty(stdout))
+            {
+                using (var reader = new StringReader(stdout))
+                {
+                    while (true)
+                    {
+                        var text = reader.ReadLine();
+                        if (text == null)
+                        {
+                            return;
+                        }
+
+                        int id;
+                        if (int.TryParse(text, out id))
+                        {
+                            childs.Add(id);
+                            // Recursively get the childs
+                            GetAllChildIdsUnix(id, childs, timeout);
+                        }
+                    }
+                }
+            }
+        }
+
+        private static void KillProcessUnix(int processId, TimeSpan timeout)
+        {
+            string stdout;
+            RunProcessAndWaitForExit(
+                "kill",
+                $"-TERM {processId}",
+                timeout,
+                out stdout);
+        }
+
+        private static int RunProcessAndWaitForExit(string fileName, string arguments, TimeSpan timeout, out string stdout)
+        {
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = fileName,
+                Arguments = arguments,
+                RedirectStandardOutput = true
```

Comment:
Can don't you need to set `UseShellExecute` to false to enable this? Also, seem like it's not even used in by methods in the class so maybe just remove it?